GVFS/GVFS.Common/X509Certificates/SystemCertificateStore.cs

```diff
@@ -0,0 +1,28 @@
+using System.Security.Cryptography.X509Certificates;
+
+namespace GVFS.Common.X509Certificates
+{
+    public class SystemCertificateStore : ICertificateStore
+    {
+        private readonly X509Store store;
+        public SystemCertificateStore()
+        {
+            this.store = new X509Store();
+        }
+
+        public void Dispose()
+        {
+            this.store.Dispose();
+        }
+
+        public X509Certificate2Collection Find(X509FindType findType, string searchString, bool validOnly)
+        {
+            if (!this.store.IsOpen)
```

Comment:
This line is causing a compilation failure on Windows (where GVFS.Common builds with .NET Framework):

```
5>------ Build started: Project: GVFS.Common, Configuration: Debug x64 ------
5>X509Certificates\SystemCertificateStore.cs(20,29,20,35): error CS1061: 'X509Store' does not contain a definition for 'IsOpen' and no extension method 'IsOpen' accepting a first argument of type 'X509Store' could be found (are you missing a using directive or an assembly reference?)
5>Done building project "GVFS.Common.csproj" -- FAILED.
```

And based on the context Window I saw in the IDE, it looks like `IsOpen` is not available in the .NET Standard API.

![image](https://user-images.githubusercontent.com/25354985/51765446-a0651480-208c-11e9-8f9c-6f5e6f0fcaa5.png)

Is there a different API you can use to perform this check?

(If you don't have a Windows machine available you can use the required PR builds to check if the build on Windows is successful)

Reply:
@wilbaker after I fixed these two, there's another issue, that `GVFS.UnitTests.Windows` uses .NET 4.6.1, which does not have `CertificateRequest` class, which I'm using in my unit tests to generate self signed certificates.  Would it be possible to target 4.7.2 in the unit tests project?

Reply:
> Would it be possible to target 4.7.2 in the unit tests project?

My preference is to leave the unit tests targeting 4.6.1 as that's what the VFS4G executables on Windows are using (and we're not mocking out everything in the .NET Framework).

Rather than retargeting `GVFS.UnitTests.Windows` what would you think of only running these tests as part of `GVFS.UnitTest`?

The cleanest way I've found for doing that is adding the following to `GitSslTests.cs`:

```
#if NETCOREAPP2_1 // .NET Framework 4.6.1 does not support System.Security.Cryptography.X509Certificates.CertificateRequest

// All of the contents of GitSslTests.cs is here and unchanged

#endif
```

I still need to take a look at the new tests you've added, and will post feedback after taking a look.  Thanks!

CC: @kewillford and @nickgra for their thoughts as well.

Reply:
After some digging through history, I agree that `#if NETCOREAPP2_1` is likely the best solution here even though it's a code smell. 

In the past when we started migrating to .NET Core, we had some tests that broke due to changes in how .NET hashes strings. While the problem was actually an issue with the way the test was written, it concerns me that we could write a UT that would behave differently than the actual product code would if they're targeting different platform versions.

In the medium to long term, I plan to move the entire project (Windows and Mac) to .NET Core 3.0 when it comes out (We need C++/CLI support on Windows, which is going to be supported in 3.0), so I can clean this up when that time comes. I think that's a better investment than us moving forward to 4.7.2 right now.

Reply:
Sounds great, will do this!